addons/pos_sale/models/sale_order.py

```diff
@@ -34,6 +34,26 @@ def _compute_amount_unpaid(self):
             total_pos_paid = sum(sale_order.order_line.filtered(lambda l: not l.display_type).mapped('pos_order_line_ids.price_subtotal_incl'))
             sale_order.amount_unpaid = sale_order.amount_total - (total_invoice_paid + total_pos_paid)
 
+    @api.depends('order_line.pos_order_line_ids')
+    def _compute_amount_to_invoice(self):
+        super()._compute_amount_to_invoice()
+        for order in self:
+            if order.invoice_status == 'invoiced':
+                continue
+            # We need to account for the downpayment paid in POS with and without invoice
+            order_amount = 0.0
+            for line in order.order_line.filtered(lambda sol: sol.is_downpayment and sol.pos_order_line_ids):
+                tax_base_line_dict = line._convert_to_tax_base_line_dict()
+                tax_base_line_dict.update({'quantity': line.qty_invoiced})
+                tax_results = self.env['account.tax'].with_company(line.company_id)._compute_taxes([
+                    tax_base_line_dict
+                ])
+                totals = next(iter(tax_results['totals'].values()))
+                order_amount += totals['amount_untaxed'] + totals['amount_tax']
+
+            order.amount_to_invoice -= order_amount
```

Comment:
Isn't using the `price_subtotal_incl` of the pos_order_line enough ?

Something like
```suggestion
            order_amount = sum(order.pos_order_line_ids.filtered(lambda pol: pol.sale_order_line_id.is_downpayment).mapped('price_subtotal_incl'))
            order.amount_to_invoice -= order_amount
```

Reply:
Maybe just `order.pos_order_line_ids.mapped('price_subtotal_incl')` would work too but I'm not sure I do not know all the workflows of sales

Reply:
Many thanks for the suggestion, it's working as expected and it's way more readable
I would keep `sum(...)` because we can add two downpayments in a single pos order